### PR TITLE
Display sampler metadata when available

### PR DESCRIPTION
**Pull request by mdcfe**
*Warning: potentially horrible React code ahead*

This is the spark-ignition counterpart to lucko/spark#58 and lucko/spark-web#4 (showing platform metadata) and also somewhat touches #1 (user/time/interval metadata).

Things to consider:
- I have no idea whether this is competent React or JSX code
- Should this be split into another "component" that can be reused in the heap summary (when implemented)?
- I haven't added any CSS for this yet. IMO it looks fine as-is, but spark-web has this metadata inside a box, and lucko/spark-web#4 makes the box expandable (hides platform by default)

Any criticism welcomed (first time doing React stuff)

### CODE REVIEW

File: src/types/Sampler.js [hunk 1]
```diff
@@ -95,4 +98,52 @@ const Name = ({ name }) => {
...
         <span className="method-part">{method}</span>
     </>
-}
\ No newline at end of file
+}
+
+// todo: where should this go? should we split this up?
+// right now it's structured very similarly to how spark-web does it
+const Metadata = ({ metadata }) => {
+    console.log(metadata)
```
Comment (astei) [inline src/types/Sampler.js]: Remove this debug code.

File: src/types/Sampler.js [hunk 2]
```diff
@@ -95,4 +98,52 @@ const Name = ({ name }) => {
...
+        if (metadata.comment) {
+            comment = ' "' + metadata.comment + '" '
+        }
+
+        const { user, startTime, interval } = metadata
+        const { type, name } = user
+        const start = new Date(startTime);
+
+        let avatarUrl = 'https://minotar.net/avatar/Console/12.png'
+        if (type === 1) {
```
Comment (astei) [inline src/types/Sampler.js]: Not a fan of the magic value going on here.

File: src/types/Sampler.js [hunk 3]
```diff
@@ -95,4 +98,52 @@ const Name = ({ name }) => {
...
-}
\ No newline at end of file
+}
+
+// todo: where should this go? should we split this up?
+// right now it's structured very similarly to how spark-web does it
+const Metadata = ({ metadata }) => {
+    console.log(metadata)
+
+    let userData
```
Comment (astei) [inline src/types/Sampler.js]: This logic can be extracted into it's own component to make this a bit neater

File: src/types/Sampler.js [hunk 4]
```diff
@@ -95,4 +98,52 @@ const Name = ({ name }) => {
...
+        }
+
+        userData = 
+            <span>
+                Profile {comment} created by <img src={avatarUrl} alt="" /> {name} at {start.toLocaleTimeString([], {hour12: true, hour: '2-digit', minute: '2-digit'})} on {start.toLocaleDateString()}, interval {interval / 1000}ms
+            </span>
+    }
+
+    let platformData
+    if (metadata.platform) {
```
Comment (astei) [inline src/types/Sampler.js:142]: Ditto

**Review by astei — CHANGES_REQUESTED** [review]
Not bad for a beginner to React

**Comment by mdcfe** [comment]
I've updated this PR according to review comments.

Currently `CommonMetadata` actually depends on metadata that is specific to the sampler output and is missing from heapdump outputs, which matches how spark-web does this but means it can't be reused for heapdumps. `PlatformMetadata` can be reused for the heapdump when implemented, but spark-web doesn't currently show any metadata for heapdumps and spark-ignition doesn't support heapdumps yet, so I'm not sure whether this is worth considering right now.

**Review by astei — APPROVED** [review]

**Comment by mdcfe** [comment]
This is ready for merge whenever you're happy to merge it.

**Pull request merged**